File: eCommerce.Models/ControllerModels/ProductVariable.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace eCommerce.Models.ControllerModels
{
    [NotMapped]
    public class ProductVariable
    {
        public int Index { get; set; }
        public SizeDetail SizeDetail { get; set; } = new SizeDetail();
        public ColorDetail ColorDetail { get; set; } = new ColorDetail();
        public List<string> ImageUrls { get; set; } = new List<string>();
        public int Price { get; set; }
        public int Discount { get; set; }
        public int Inventory { get; set; }
        public bool ImageShow { get; set; }

    }
}

File: eCommerce.Mobile/ecommerce/lib/models/productvariable.dart

import 'package:ecommerce/models/sizedetail.dart';
import 'package:ecommerce/models/colordetail.dart';

class ProductVariable{
  ProductVariable();

  int index = 0;
  SizeDetail sizeDetail = SizeDetail();
  ColorDetail colorDetail = ColorDetail();
  List<String> imageUrls = [];
  int price = 0;
  int discount = 0;
  int inventory = 0;
  bool imageShow = false;
}